### PR TITLE
Fix offline diarization pipeline producing single-speaker output

### LINKED ISSUE #513: OfflineDiarizerManager fails to indentify speakers
Hi,

I'm seeking some help, likely with configuration. OfflineDiarizerManager is failing to correctly identify different speakers. I'm not sure what do do. I attach a small clip from a podcast. Regardless of the settings I've tried, OfflineDiarizerManager continues to return 30+ segments with only one speaker. Curiously, the last speaker is marked as "S2". I have also performed diarization on an extended version (+ 1 hour) and still with nearly the same results, a failure to correctly identify speakers.

Sortformer does a nearly perfect job, but is impossibly slow to load and process (especially on iOS).

I'm quite stuck and hoping for some help. Thanks.

[Matan_Trim_1080.m4a.zip](https://github.com/user-attachments/files/26648477/Matan_Trim_1080.m4a.zip)

### PR DESCRIPTION
**Pull request by thechatk**
## Summary

Fixes three bugs in the offline diarization pipeline that caused it to attribute nearly all segments to a single speaker:

- **`vDSP_mtrans` dimension swap** — `frameCount` and `speakerCount` arguments were reversed, corrupting transposed speaker masks to be nearly identical across all speakers
- **Missing activity ratio filter** — the reference pyannote implementation filters speakers with <20% clean activation, but the current code only filters completely silent speakers, allowing junk embeddings through to clustering
- **Soft masks vs binary masks** — the reference derives per-frame speaker masks from argmax on powerset logits (binary 0/1), but the current code uses soft probabilistic masks via matrix-vector multiplication, producing blurred activations

## Test results

After all three fixes, diarization works correctly across multiple test scenarios. Testing against a 467-second 3-speaker audio file achieved **97% F1 score** vs PyTorch pyannote while maintaining real-time performance (120x faster than real-time).

## Files changed

- `Sources/FluidAudio/Diarizer/Offline/Extraction/OfflineEmbeddingExtractor.swift`
- `Sources/FluidAudio/Diarizer/Offline/Segmentation/OfflineSegmentationProcessor.swift`

Fixes #513

---

<a href="https://app.devin.ai/review/fluidinference/fluidaudio/pull/523" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by Alex-Wengg** [comment]
after running the 16 AMI testing files
```
  ┌─────────────────┬───────────────────┬───────────────┐
  │     Branch      │      Avg DER      │   Avg RTFx    │
  ├─────────────────┼───────────────────┼───────────────┤
  │ Main (buggy)    │            20.74% │         55.9x │
  ├─────────────────┼───────────────────┼───────────────┤
  │ PR #523 (fixed) │            10.62% │         94.7x │
  ├─────────────────┼───────────────────┼───────────────┤
  │ Δ               │ −10.12pp (−48.8%) │ +38.8x (+69%) │
  └─────────────────┴───────────────────┴───────────────┘
```
verified benchmark improvements
